internal/pkg/utils/testproject/snapshot.go

```diff
@@ -186,14 +188,26 @@ func (p *Project) NewSnapshot() (*fixtures.ProjectSnapshot, error) {
 	grp.Go(func() error {
 		// Files metadata are not atomic, wait a moment.
 		// The creation/deletion of the file does not take effect immediately.
-		time.Sleep(100 * time.Millisecond)
-		return p.keboolaProjectAPI.
-			ListFilesRequest().
-			WithOnSuccess(func(_ context.Context, apiFiles *[]*keboola.File) error {
-				files = *apiFiles
-				return nil
-			}).
-			SendOrErr(ctx)
+		// We accept the result if it is 10x the same.
+		for attempt := 0; attempt < 10; attempt++ {
+			time.Sleep(100 * time.Millisecond)
+			err := p.keboolaProjectAPI.
+				ListFilesRequest().
+				WithOnSuccess(func(_ context.Context, apiFiles *[]*keboola.File) error {
+					// Reset the counter, if results differs.
+					// Ignore the URL field, it may contain a random/volatile time-based signature.
+					if files != nil && !cmp.Equal(&files, apiFiles, cmpopts.IgnoreFields(keboola.File{}, "URL")) {
+						attempt = 0
+					}
+					files = *apiFiles
+					return nil
+				}).
+				SendOrErr(ctx)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
```

Comment:
Casto mi padaju testy ohladom `files`.

Pridal som, ze musime `10x` po sebe ziskat rovnaku odpoved aby sme to brali ako platny stav.
URL obsahuje `AWS-Signature` a ta je zalozena na aktualnom timestamp, takze sa musi ignorovat.